macos/ReactTestApp/AppDelegate.swift

```diff
@@ -41,24 +45,34 @@ final class AppDelegate: NSObject, NSApplicationDelegate {
             return
         }
 
-        manifest.components.enumerated().forEach { index, component in
-            let title = component.displayName ?? component.appKey
-            let item = reactMenu.addItem(
-                withTitle: title,
-                action: #selector(onComponentSelected),
-                keyEquivalent: index < 9 ? String(index + 1) : ""
+        let components = manifest.components ?? []
+        if components.isEmpty {
+            NotificationCenter.default.addObserver(
+                forName: .ReactTestAppDidRegisterApps,
+                object: nil,
+                queue: .main,
+                using: { [weak self] note in
+                    guard let strongSelf = self,
+                          let appKeys = note.userInfo?["appKeys"] as? [String]
+                    else {
+                        return
+                    }
+
+                    let components = appKeys.map { Component(appKey: $0) }
+                    strongSelf.onComponentsRegistered(components, enable: true)
+                    if components.count == 1, !strongSelf.isPresenting {
+                        strongSelf.present(components[0])
+                    }
+                }
             )
-            item.tag = index
-            item.keyEquivalentModifierMask = [.shift, .command]
-            item.isEnabled = false
-            item.representedObject = component
         }
 
-        let components = manifest.components
+        onComponentsRegistered(components, enable: false)
```

Comment:
If there is a list of componetns in the manifest, wouldn't you want to call this with enable: true?

Reply:
The property is for deciding whether the menu item should be enabled. Since the bridge hasn't been initialized yet, we should prevent them from being clicked. They get enabled when it is:

https://github.com/microsoft/react-native-test-app/blob/fae8b9fd5078e6c0d2d2cf0b6e94a29f58e1e712/macos/ReactTestApp/AppDelegate.swift#L83